backend/entities/courses/user_section_entity.py

```diff
@@ -31,12 +33,29 @@ class UserSectionEntity(EntityBase):
     # User for the current relation
     # NOTE: This is ultimately a join table for a many-to-many relationship
     user_id: Mapped[int] = mapped_column(ForeignKey("user.id"), primary_key=True)
+    user: Mapped["UserEntity"] = relationship(back_populates="sections")
 
     # Section for the current relation
     # NOTE: This is ultimately a join table for a many-to-many relationship
     section_id: Mapped[int] = mapped_column(
         ForeignKey("courses__section.id"), primary_key=True
     )
+    section: Mapped["SectionEntity"] = relationship(back_populates="members")
```

Comment:
Let's specify the `section_id` ahead of the `user_id`. This is a minor optimization, and perhaps overkill, but the brief justification is typically the arrangement of rows on disk will be in the default ascending ordering of their primary keys. For compound primary keys, this means the first column is the most important. When it comes to joining, the larger joins are likely going to be going from sections -> users rather than user -> sections (where most undergrads may have a max of like... 8 semesters * 5 sections = 40 classes but most will not have any association with CS, but a section like 423 will have 180 users associated). Having this physical colocation means much quicker reads when grabbing in bulk.

Reply:
Interesting! Sounds good to me 👍